pipeline-yaml_5b5744d40428631d7a8940a9.md

```diff
@@ -0,0 +1,548 @@
+  * [About Pipeline YAML](#pipeline-yaml-reference)
+  * [The preamble](#the-preamble)
+    * [version](#version)
+    * [name](#name)
+    * [agent](#agent)
+    * [Preamble example](#preamble-example)
+  * [Blocks](#blocks)
+  * [Task](#task)
+  * [Example of task](#example-of-task)
+  * [Jobs](#jobs)
+  * [Example of jobs](#example-of-jobs)
+  * [The secrets property](#the-secrets-property)
+  * [Example of secrets](#example-of-secrets)
+  * [Complete examples](#complete-configuration-examples)
+  * [The order of execution](#the-order-of-execution)
+  * [See also](#see-also)
+
+
+# Pipeline YAML Reference
+
+This document is the reference for the YAML grammar used for describing the
+pipelines of Semaphore 2.0.
+
+The core properties of a Semaphore 2.0 pipeline configuration file are `blocks`,
+which appear only once at the beginning of a YAML file, `task`, which can
+appear multiple times and `jobs`, which can also be repeated.
+
+
+#### Comments
+
+Lines that begin with `#` are considered comments and are being ignored by the
+YALM parser, which is not a Semaphore 2.0 feature but the way YAML files work.
+
+
+## The preamble
+Each Semaphore 2.0 pipeline configuration file has a preamble, which is
+mandatory and should be present on every `.semaphore/semaphore.yml` file you create.
+
+### Properties
+
+### version
+
+The `version` property is a string value that shows the version of the
+pipeline YAML file that will be used.
+
+List of valid values: `"v1.0"`
+
+Example: `version: "v1.0"`
+
+### name
+
+The `name` property is a Unicode string that assigns a name to the Semaphore
+2.0 project and is optional.
+
+*Caution*: The `name` property can be found in other sections because it
+is used for defining the name of a job inside a `jobs` block or the name of
+a `task` section defined using `task`.
+
+Example: `name: "The name of the Semaphore 2.0 project"`
+
+
+### agent
```

Comment:
I see `agent` on the same level as `blocks` and `switch` and not as part of preamble. Also `agent` can also be part of task. Agent is required part of pipeline file as are `blocks`.

```yaml
version: "v1.0"          # <- required
name: Something    # <- required

agent:                       # <- required

blocks:                     # <- required

switch:                     # <- optional
```

I would love to hear some more computer science friendly term that could serve as a substitute for `preamble`. I don't remember hearing those in CS speak.

Reply:
What about `preface` and `prelude`? Personally, I prefer `preface`.